examples/demo/demo.ts

```diff
@@ -164,8 +165,13 @@ const appActions = {
   ): Promise<Room | undefined> => {
     const room = new Room(roomOptions);
 
+    const tokenSource = new TokenSource.Literal({
```

Comment:
seems cleaner to have a factory method instead.

also we should use camelCase in all JS types. that's more native here.

```suggestion
    const tokenSource = TokenSource.staticToken({
       url,
       token,
    })
```

Reply:
+1 for the factory method. 

Discussed with Ryan that ideally we'd want to use protobuf's `toJson` internally here which gives us
- single source of truth for clients
- language idiomatic casing styles
- consistent on the wire format (snake case'd JSON as defined by Tobias)

Reply:
> seems cleaner to have a factory method instead.

The reason why I did it this way over an explicit factory constructor type pattern was that fundamentally it's challenging to model all the `TokenSource` functionality as one class. There's too much breadth in functionality for a single monolithic class to work in all cases. Given this, it makes sense to split it up across a few levels of abstraction like I have done now. That inevitably means you end up with `TokenSourceLiteral`, `TokenSourceCustom`, `TokenSourceSandboxTokenServer`, etc. 

The question becomes then how these should be exposed to a user. What I did here with a namespace I thought was fairly elegant, and seemed reminiscent to me of [dot notation in react component names](https://legacy.reactjs.org/docs/jsx-in-depth.html#using-dot-notation-for-jsx-type) (like `<MyContext.Provider value={...}>`), but I totally understand it could be seen as maybe a little weird / unconventional. 

Assuming the existing pattern I have isn't preferable, I see two other realistic options:
- **Option 1**: Make the existing `TokenSource` implementations all private, and add factory static methods similar to what you have proposed. In other words, at a 10k foot view, something like this:
```tsx
export class TokenSource {
  /* ... implementation code here ... */
  
  /** "literal" doc string here */
  static literal(request: TokenSourceRequest) {
    return new TokenSourceLiteral(request);
  }
  /** "custom" doc string here */
  static custom(fn: (request: TokenSourceRequest) => Promise<TokenSourceResponse>) {
    return new TokenSourceCustom(fn);
  }
  /* ... etc ... */
}

/** copy of "literal" doc string here (ideally) */
class TokenSourceLiteral extends TokenSource { /* ... */ }
/** copy of "custom" doc string here (ideally) */
class TokenSourceCustom extends TokenSource { /* ... */ }
// ... etc ...

// Then, to use it later on:
import { TokenSource } from 'livekit-client';
const tokenSource = TokenSource.literal({ /* ... */ });
```

The biggest con to be of the above is just all the duplication / boilerplate through all the levels of indirection (especially the doc strings... I guess it's not _necessarily_ required to duplicate them, but it sucks that we wouldn't have those on the internal abstraction levels), but maybe that's worth for the user it if this is really the best interface.

- **Option 2**: Get rid of the namespace and expose all classes as separate imports all starting with the same prefix. This gets rid of the leading namespace with the dot and embraces fully everything just being a class. So for example:
```tsx
export class TokenSource {
  /* ... implementation code here ... */
}

/** "literal" doc string here */
export class TokenSourceLiteral extends TokenSource { /* ... */ }
/** "custom" doc string here */
export class TokenSourceCustom extends TokenSource { /* ... */ }
// ... etc ...

// Then, to use it later on:
import { TokenSourceLiteral } from 'livekit-client';
const tokenSource = new TokenSourceLiteral({ /* ... */ });
```

While it is verbose, the nice thing about this approach is the lack of duplication - no multiple sets of function args to keep up to date, only one doc string, etc. Also another nice thing is anybody else who wanted to make their own token source could just make a class starting with TokenSource and it matches with the stock implementations.

-----------

I generally am a fan of what I have now, but I'm curious where you / others like @lukasIO weigh in between what I have currently, option 1, and option 2 given all the pros and cons I outlined.

Reply:
I prefer option 1.. it's verbose to copy the docstring once.. but for a end user it's so much nicer.. I don't have to think about these types to import. For most this could also be typing `TokenSource.` and seeing the completion on the options on how to create it.. that in itself makes the system a lot more intuitive.